invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/components/ExportSearchResults/ExportSearchResults.js

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Qs from 'qs';
+import { Button, Popup, Dropdown, Menu } from 'semantic-ui-react';
+import {
+  InvenioRequestSerializer,
+  withState as withSearchState,
+} from 'react-searchkit';
+import { invenioConfig } from '../../../../common/config/invenioConfig';
+
+/** Simple component rendering a small dialog to choose format and page of results to export. */
+class ExportDialog extends Component {
+  constructor(props) {
+    super(props);
+    this.max = invenioConfig.max_results_window - 1;
+    const totalHits = this.props.total;
+    const lastPage = totalHits % this.max > 0 ? 1 : 0;
+    const totalPages = Math.floor(totalHits / this.max) + lastPage;
+
+    this.formatOptions = [
+      {
+        key: 'csv',
+        text: 'CSV',
+        value: 'csv',
+      },
+      {
+        key: 'json',
+        text: 'JSON',
+        value: 'json',
+      },
+    ];
+
+    this.resultsOptions = Array.from(Array(totalPages).keys()).map(i => {
+      const page = i + 1;
+      // build dropdown options, e.g.:
+      // 1 - 9999, 10000 - 19999, ...
```

Comment:
If the display limit is `10000` shouldn't it say `1-10000` instead of `1-9999`? Or write `0-9999` then `10000-19999`?

Reply:
Nice catch! The thing is that I believe there is an issue in `invenio-records-rest`: if I set a size of `10000`, it fails throwing an error that the limit has been reached.
I think there is a < 10000 instead of <= 10000.
So for now I have to keep it like this unless we manage to fix that issue.